File: hw/sanchez/u3/hw28Strategy/StrategyPython/controller/sorting_context.py
from model.bubble_sort import BubbleSort
from model.insertion_sort import InsertionSort
from model.quick_sort import QuickSort


class SortingContext:

    def __init__(self):
        self.sorting_strategy = None

    def sort(self, data):
        self.sorting_strategy = self.set_sort_strategy(len(data))
        return self.sorting_strategy.sort(data)

    def set_sort_strategy(self, size):

        if 0 < size < 30:
            self.sorting_strategy = BubbleSort()

        elif 30 <= size < 100:
            self.sorting_strategy = InsertionSort()

        else:
            self.sorting_strategy = QuickSort()

        return self.sorting_strategy

File: hw/sanchez/u3/hw28Strategy/StrategyJava/Strategy/src/main/java/ec/edu/espe/strategy/controller/SortingContext.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.strategy.controller;

import ec.edu.espe.strategy.model.BubbleSort;
import ec.edu.espe.strategy.model.InsertionSort;
import ec.edu.espe.strategy.model.QuickSort;
import ec.edu.espe.strategy.model.SortingStrategy;

/**
 *
 * @author Joel Sanchez, The Softwarriors, @ESPE
 */
public class SortingContext {

    private SortingStrategy sortingStrategy;

    public int[] sort(int[] data) {
        sortingStrategy = setSortStrategy(data.length);
        return sortingStrategy.sort(data);
    }

    public SortingStrategy setSortStrategy(int size) {

        if (size > 0 && size < 30) {
            sortingStrategy = new BubbleSort();
        } else if (size >= 30 && size < 100) {
            sortingStrategy = new InsertionSort();
        } else {
            sortingStrategy = new QuickSort();
        }

        return sortingStrategy;
    }

}
